tests/unit/test_dbapi_connection.py

```diff
@@ -41,29 +41,26 @@ def _mock_client(self):
         return mock_client
 
     def _mock_bqstorage_client(self):
-        if bigquery_storage_v1 is None:
-            return None
-        mock_client = mock.create_autospec(
-            bigquery_storage_v1.client.BigQueryReadClient
-        )
-        mock_client.transport = mock.Mock(spec=["channel"])
-        mock_client.transport.channel = mock.Mock(spec=["close"])
+        # Assumption: bigquery_storage exists. It's the test's responisbility to
+        # not use this helper or skip itself if bqstroage is not installed.
+        mock_client = mock.create_autospec(bigquery_storage.BigQueryReadClient)
+        mock_client._transport = mock.Mock(spec=["channel"])
+        mock_client._transport.grpc_channel = mock.Mock(spec=["close"])
         return mock_client
 
     def test_ctor_wo_bqstorage_client(self):
         from google.cloud.bigquery.dbapi import Connection
 
         mock_client = self._mock_client()
-        mock_bqstorage_client = self._mock_bqstorage_client()
-        mock_client._create_bqstorage_client.return_value = mock_bqstorage_client
+        mock_client._create_bqstorage_client.return_value = None
```

Comment:
Maybe create a sentinel `object()` here? I think we want to ensure that a BQ Storage client is created by default.

Reply:
I'll see if a sentinel mock will work out of the box, because IIRC there is an "is None" check in the DB API code, thus we need to actually use `None`.

Reply:
Checked the test, and the scenario it covers is when the `bqstorage` extra is not installed. The BigQuery client's factory method returns `None` in this case, hence the `None` return value of the mocked factory.